File: java/src/main/java/parser/ast/Variable.java
package parser.ast;

import lexer.Token;

public class Variable extends Factor {

    private Token typeLexeme = null;

    public Variable(Token token) {
        super(token);
        this.type = ASTNodeTypes.VARIABLE;
    }

    public void setTypeLexeme(Token token) {
        this.typeLexeme = token;
    }
    public Token getTypeLexeme(){
        return this.typeLexeme;
    }
}

File: TinyScript/js/src/parser/ast/Variable.js
const ASTNodeTypes = require('./ASTNodeTypes')
const {Factor} = require('./index')

class Variable extends Factor{


    constructor(token){
        super(token)
        this.type = ASTNodeTypes.VARIABLE
        this.typeLexeme = null
    }


    setTypeLexeme(lexeme){
        this.typeLexeme = lexeme
    }

    getTypeLexeme(){
        return this.typeLexeme
    }



}

module.exports = Variable